### PR TITLE
add the DAO to check incomplete workflow in agent

### PR DESCRIPTION
**Pull request by ticoann**
For checking draining status, use this for the first check if this passes check the DBS, PhEDEx upload status.

### CODE REVIEW

File: src/python/WMComponent/DBS3Buffer/DBSBufferUtil.py [hunk 1]
```diff
@@ -240,3 +240,11 @@ def summaryPhEDExDBSStatus(self, data):
             else:
                 summary[workflow]["DBSUploaded"] = False
         return summary
+
+    def isAllWorkflowCompleted(self):
+        """
+        check whether all the workflows are completed status in give agent (not globaly in reqmgr)
+        This should be used as preconditon for checking draining status
+        """
+        completeFWFlag = self.daoFactory(classname="isAllWorkflowsCompleted")
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/DBSBufferUtil.py]: dao name is wrong

File: src/python/WMComponent/DBS3Buffer/Oracle/IsAllWorkflowsCompleted.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from __future__ import print_function, division
+"""
+_IsAllWorkflowsCompleted_
+
+Oracle implementation of DBS3Buffer.IsAllWorkflowsCompleted
+"""
+
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/Oracle/IsAllWorkflowsCompleted.py]: no need for all these blank lines :)

File: src/python/WMComponent/DBS3Buffer/Oracle/IsAllWorkflowsCompleted.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+
+from WMComponent.DBS3Buffer.MySQL.IsAllWorkflowsCompleted \
+    import IsAllWorkflowsCompleted as MySQLIsAllWorkflowsCompleted
+
+class IsAllWorkflowsCompleted(MySQLIsAllWorkflowsCompleted):
+    """
+    Oracle version
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/Oracle/IsAllWorkflowsCompleted.py]: you probably forgot to close this docstring and pass this class

**Review by amaltaro — CHANGES_REQUESTED** [review]
A few minor issues need to be addressed, otherwise the code looks good.

However, I think we should make this DAO a bit more generic to avoid creating gazillions of DAOs for the draining process. I thought we could make it return two different results:
1) just a boolean flag, as it is already implemented
2) return distinct names - for cases we want to know further what are the workflows still holding us on this draining.

So we could replace that SQL query by
`select distinct name from dbsbuffer_workflow WHERE completed=0;`
what do you think?

**Comment by ericvaandering** [comment]
test this please

**Comment by ticoann** [comment]
included in #7926

**Pull request closed without merging**